tools/codegen/api/types.py

```diff
@@ -201,17 +188,31 @@ def _from_grouped_arguments(
             # principle, we should be able to do this at some later
             # point in time with other overload disambiguation
             argument_packs = tuple(
-                cpp.argument_faithful(a).no_default() for a in arguments
+                cpp.argument_faithful(a).no_default() for a in (tuple(positional_and_kwargs) + self.func.arguments.out)
             )
         else:
             argument_packs = tuple(
-                cpp.argument(a) for a in arguments
+                cpp.argument(a) for a in (self.func.arguments.out + tuple(positional_and_kwargs))
             )
-        return CppSignature(
-            func=func,
-            _argument_packs=argument_packs,
-            _returns_type=cpp.returns_type(func.returns),
-        )
+        return argument_packs
+
+    def name(self) -> str:
+        return cpp.name(self.func, use_suffix_for_out_overloads=not self.faithful)
```

Comment:
Maybe it would be more direct and clear to just replace `use_suffix_for_out_overloads` with `faithful`.